pkg/operator/sync_test.go

```diff
@@ -118,3 +123,242 @@ func TestWaitForDeploymentRollout(t *testing.T) {
+			exisingWebhook: func() *admissionregistrationv1.ValidatingWebhookConfiguration {
+				webhook := defaultWebhook.DeepCopy()
+				webhook.Webhooks = append(webhook.Webhooks, mapiv1.MachineValidatingWebhook())
+				return webhook
+			},
+		},
+		{
+			testCase:     "Edge case: It shoud not update webhooks if some list inside webhooks configuration was extended with an element",
+			expectUpdate: false,
+			exisingWebhook: func() *admissionregistrationv1.ValidatingWebhookConfiguration {
+				webhook := defaultWebhook.DeepCopy()
+				webhook.Webhooks[0].Rules[0].Operations = append(webhook.Webhooks[0].Rules[0].Operations, admissionregistrationv1.Connect)
+				return webhook
+			},
+		},
+		{
+			testCase:     "Edge case: But it should update webhooks if some list inside webhooks configuration had a change in the order",
+			expectUpdate: true,
+			exisingWebhook: func() *admissionregistrationv1.ValidatingWebhookConfiguration {
+				webhook := defaultWebhook.DeepCopy()
+				rules := []admissionregistrationv1.OperationType{admissionregistrationv1.Connect}
+				webhook.Webhooks[0].Rules[0].Operations = append(rules, webhook.Webhooks[0].Rules[0].Operations...)
+				return webhook
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.testCase, func(t *testing.T) {
+			stop := make(chan struct{})
+			defer close(stop)
+			optr := newFakeOperator(nil, nil, stop)
+			optr.syncHandler = nil
+			client := optr.kubeClient.AdmissionregistrationV1().ValidatingWebhookConfigurations()
+
+			expectedWebhook, err := client.Create(context.Background(), expectedConfiguration, metav1.CreateOptions{})
+			if err != nil {
+				t.Fatalf("Unexpected error during creation of an expected webhook configuration: %q", err.Error())
+			}
+			if tc.exisingWebhook != nil {
+				if _, err = client.Create(context.Background(), tc.exisingWebhook(), metav1.CreateOptions{}); err != nil {
+					t.Fatalf("Unexpected error during creation of an exising webhook configuration: %q", err.Error())
+				}
+			}
+			defer func() {
+				if err = client.Delete(context.Background(), expectedConfiguration.Name, metav1.DeleteOptions{}); err != nil {
+					t.Fatalf("Unexpected error during deletion of an expected webhook configuration: %q", err.Error())
+				}
+				if err = client.Delete(context.Background(), defaultWebhook.Name, metav1.DeleteOptions{}); err != nil {
+					t.Fatalf("Unexpected error during deletion of an exising webhook configuration: %q", err.Error())
+				}
+			}()
+
+			if !cache.WaitForCacheSync(stop, optr.validatingWebhookListerSynced) {
+				t.Fatalf("Failed to sync caches")
+			}
+			if err := optr.syncValidatingWebhook(); err != nil {
+				t.Fatalf("Unexpected error during 'syncValidatingWebhook': %q", err.Error())
+			}
+
+			existingWebhook, err := client.Get(context.Background(), defaultWebhook.Name, metav1.GetOptions{})
+			if err != nil {
+				t.Fatalf("Unexpected error while getting validating webhook: %q", err.Error())
+			}
+
+			if tc.expectUpdate {
```

Comment:
it's no clear to me how we use `expectUpdate`. Aren't this expectations equally true for the case we don't expect an update?